### PR TITLE
Refine Helm Charts: Add missing namespaces.

### PR DESCRIPTION
**Pull request by yusuoh**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -48,6 +48,10 @@ install/kubernetes/istio-sidecar-injector-configmap-release.yaml
 install/kubernetes/istio-sidecar-injector.yaml
 install/kubernetes/istio.yaml
 install/kubernetes/istio-all.yaml
+install/kubernetes/orig_istio-auth.yaml
```
Comment (costinm) [inline .gitignore]: Can we generate files under $OUT and not in the source tree ?

**Comment by costinm** [comment]
Adding more people - I don't think we have tiller tests - or if it works with tiller.

We need to keep it working for both

**Comment by sdake** [comment]
@costin the helm charts worked with tiller a few weeks ago.  I'll test this PR later tonight against tiller - I've got a few errands to run.  I don't think there  is harm in hard-coding the namespaces, although typically with helm -n argument is used to specify the namespace (and is automatically set in that condition IIRC).  I could be wrong on th eabove - the helm charts went through a large refactor.

I'll report back this evening.  otherwise lgtm

**Comment by mandarjog** [comment]
Why do we need namespace embedded in the artifacts ?
kubectl -n < > apply
or helm -n  work just fine ..

What is the helm best practice here?

**Comment by yusuoh** [comment]
This change is to make sure that the generated yamls from Istio release are correct (i.e. user can use kubectl apply -f istio.yaml without specifying namespaces). Currently some of them are tagged with namespaces and some are not. The kubectl -n and helm -n should still work as before. 

It is not breaking the e2e tests only because those test are always using kubectl -n to set up the environment.

**Comment by yusuoh** [comment]
Btw, it seems that helm install -n and Helm template -n have different behavior. The previous one will tag all resources with correct namespace while the later won't unless you specifiy {{.Release.Namespace}} explicitly.

**Comment by yusuoh** [comment]
There is a related issue in Helm: https://github.com/kubernetes/helm/issues/3553

**Comment by costinm** [comment]
I think we should follow Helm practices - if it means 'kubectl apply -n istio-system' - it's fine with me.

If helm recommends including the namespace in template - good with me as well.

**Comment by yusuoh** [comment]
It seems that a new file is added by previous PR #4001 (install/kubernetes/helm/istio/templates/namespace.yaml), which broke the helm install. Because helm -n is trying to create namespace which already exits.

istio$ helm install install/kubernetes/helm/istio --namespace=istio-system
Error: release dangling-buffalo failed: namespaces "istio-system" already exists

After I deleted that file, it passes with this PR. However, we need to find a way to generate yaml from templates correctly.

**Comment by yusuoh** [comment]
Fixed the above namespace issue in #4612 by moving the namespace file out of helm chart.

**Comment by sdake** [comment]
Apologies I failed to test this PR last night.  Sounds like you gave it a spin with Tiller though @yusuoh.  I think Helm best practice is to use -n to override namespaces.

CCing @jascott1 who is a helm core and can answer the best practice question.

**Comment by jascott1** [comment]
Having a Namespace manifest is currently problematic with Helm as you have discovered. Just using `-n` is more idiomatic.

**Comment by yusuoh** [comment]
@jascott1 This PR is not adding the namespace manifest but only to specify the metata.namespace field in k8s resources. Do you think it is OK to do that?

**Comment by jascott1** [comment]
BTW I corrected my above post, just for clarity its `--namespace` for namespace and `-n` is release name.

**Pull request merged**